common/src/main/java/com/wynntils/models/raid/RaidModel.java

```diff
@@ -221,6 +250,22 @@ public RaidKind getCurrentRaid() {
         return currentRaid;
     }
 
+    public List<String> getRaidBuffMajorIds(String playerName) {
+        List<String> buffNames = partyRaidBuffs.getOrDefault(playerName, Collections.emptyList());
+        if (buffNames.isEmpty()) return buffNames;
+
+        List<String> majorIds = new ArrayList<>();
+
+        for (String buffName : buffNames) {
+            String majorId = this.currentRaid.majorIdFromBuff(buffName);
+            if (majorId == null) continue;
+
+            majorIds.add(majorId);
+        }
+
+        return majorIds;
+    }
```

Comment:
```suggestion
        if (!partyRaidBuffs.containsKey(playerName)) return List.of();

        List<String> buffNames = partyRaidBuffs.get(playerName);
        List<String> majorIds = new ArrayList<>();

        for (String buffName : buffNames) {
            String majorId = this.currentRaid.majorIdFromBuff(buffName);
            if (majorId == null) continue;

            majorIds.add(majorId);
        }

        return majorIds;
    }
```

I am not sure if this was an attempt of (premature) optimization, or you just coded it this way, but simple code is always better when performance is not an issue. 

You can optionally try to rewrite this using streams, so something like `playerRaidBuffs.stream().map(...)...`

Reply:
I'm happy with the refactor you suggested and have refactored that. I did just write it that way.

If streams is the convention that you'd like to follow (as I saw in the range visualizer itself) then I can do so but I think that there's no benefit to using them. With such little data streams are slower and a simple for loop is usually more readable too.